### ISSUE TITLE
att问题

### ISSUE THREAD

**Issue by caikw0602** [Lilidamowang/CoTNet-50#1]
楼主好，cotblock中在求k2时，没有用att和v相结合的办法来求，也就是v和最后的结果是不相关的，这是否会有问题呢？

**Comment by Lilidamowang** [comment]
你好，在文件第54行，对att进行softmax操作后，按照论文中的结构图所示直接与Value进行了相乘，V影响了最后的输出结果。我在知乎中有一篇讲解文章，希望对你有帮助。https://www.zhihu.com/column/p/396946811

**Comment by caikw0602** [comment]
感谢您的回复，是我看差了，不好意思。在这里想请教一个问题，您是采用att进行softmax操作后与v相乘的方式来获得临近上下文信息的，源代码里面是采用aggregation_zeropad函数来实现这一步的，请问您看源代码的这部分了吗？应该怎么理解这个函数呢？有点没看懂

**Comment by hpz4311** [comment]
你好，不知道您有没有看官方的实现代码，想请教两个问题
1.在得到k1时，官方代码里进行了一个lacal_conv操作，最终调用了aggregation_zeropad来实现，不太懂这一步
2.好像最终融合时候官方代码不是简单的进行k1+k2,而是下面这样
        B, C, H, W = x.shape
        x = x.view(B, C, 1, H, W)
        k = k.view(B, C, 1, H, W)
        x = torch.cat([x, k], dim=2)

        x_gap = x.sum(dim=2)
        x_gap = x_gap.mean((2, 3), keepdim=True)
        x_attn = self.se(x_gap)
        x_attn = x_attn.view(B, C, self.radix)
        x_attn = F.softmax(x_attn, dim=2)
        out = (x * x_attn.reshape((B, C, self.radix, 1, 1))).sum(dim=2)

**Comment by Chang-shuaishuai** [comment]
> 好像最终融合时候官方代码不是简单的进行k1+k2,而是下面这

请问这个问题，你想明白了吗？？我也遇到了，很奇怪
